# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/xpcprivate.h
# Commit: 930054aa138b
# Full Hash: 930054aa138bd32f51bbe2ec84e07267859afa6e
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2018-11-29 04:28:09
# Description:
#   Bug 1508102 - Take account of XPCWrappedJS::mJSObjGlobal in cycle collector methods r=mccr8 a=abillings
# ==============================================================================

diff -r 037b1d961ac5 -r 930054aa138b js/xpconnect/src/xpcprivate.h
--- a/js/xpconnect/src/xpcprivate.h	Wed Nov 28 19:23:08 2018 +0200
+++ b/js/xpconnect/src/xpcprivate.h	Wed Nov 28 17:29:19 2018 +0000
@@ -1852,6 +1852,10 @@
      */
     JSObject* GetJSObjectPreserveColor() const { return mJSObj.unbarrieredGet(); }
 
+    JSObject* GetJSObjectGlobalPreserveColor() const {
+        return mJSObjGlobal.unbarrieredGet();
+    }
+
     // Returns true if the wrapper chain contains references to multiple
     // compartments. If the wrapper chain contains references to multiple
     // compartments, then it must be registered on the XPCJSContext. Otherwise,
